packages/table/components/Table.tsx

```diff
@@ -0,0 +1,169 @@
+import * as React from "react";
+import { css } from "emotion";
+import { AutoSizer, MultiGrid, GridCellProps } from "react-virtualized";
+import { IColumnProps, Column } from "./Column";
+import memoizeOne from "memoize-one";
+import { coreColors as Colors } from "../../shared/styles/color";
+
+export interface ITableProps {
+  /**
+   * Contains all data represented in the table.
+   */
+  data: any[];
+  children:
+    | Array<React.ReactElement<IColumnProps>>
+    | React.ReactElement<IColumnProps>;
+}
+
+const ROW_HEIGHT = 35;
+
+const DEFAULT_WIDTH = 1024;
+const DEFAULT_HEIGHT = 768;
+export class Table<T> extends React.PureComponent<ITableProps, {}> {
+  public multiGridRef: { recomputeGridSize?: any } = {};
+
+  public getData = memoizeOne(
+    (data: T[]): Array<{}> | T[] => {
+      return [{}, ...data];
+    }
+  );
+
+  public getColumnSizes = memoizeOne(
+    (
+      children: Array<React.ReactElement<IColumnProps>>,
+      width: number
+    ): number[] => {
+      const totalColumns: number = children.length;
+      let remainingWidth: number = width;
+      return children.map((child, currentIndex) => {
+        const calculatedWidth = child.props.width({
+          width,
+          totalColumns,
+          currentIndex,
+          remainingWidth
+        });
+        remainingWidth -= calculatedWidth;
+        return calculatedWidth;
+      });
+    }
+  );
+
+  constructor(props) {
+    super(props);
+
+    this.getCell = this.getCell.bind(this);
+    this.setRef = this.setRef.bind(this);
+    this.updateGridSize = this.updateGridSize.bind(this);
+    this.getGrid = this.getGrid.bind(this);
+  }
+
+  public render() {
+    return (
+      <AutoSizer
+        defaultWidth={DEFAULT_WIDTH}
+        defaultHeight={DEFAULT_HEIGHT}
+        onResize={this.updateGridSize}
+      >
+        {this.getGrid}
+      </AutoSizer>
+    );
+  }
+
+  private getGrid({ width, height }) {
+    const mergedData = this.getData(this.props.data);
+
+    const columnCount = React.Children.count(this.props.children);
+    const columnSizes = this.getColumnSizes(
+      React.Children.toArray(this.props.children) as Array<
+        React.ReactElement<IColumnProps>
+      >,
+      width
+    );
+
+    const getColumnSize = ({ index }) => columnSizes[index];
```

Comment:
We may wanna replace `=>` with regular function
```js
function getColumnSize({ index }) { 
  const columnCount = React.Children.count(this.props.children);
  const columnSizes = this.getColumnSizes(
     React.Children.toArray(this.props.children) as Array<
       React.ReactElement<IColumnProps>
     >,
     width
  );
  return columnSizes[index]
}
```

extract the whole logic it needs for `columnSizes` and bind it in the constructor 
so that we spare some performance. Otherwise `=>` will be bound on every function call and a new function will be created that will lead to re-render of `MultiGrid`

Just replacing the line `const getColumnSize = ({ index }) => columnSizes[index];`
with `function getColumnSize ({ index }) { return columnSizes[index]; }` could work.

Reply:
I am not sure about the rerender part. Since the column width is getting cached. However, there is nothing speaking against pulling it out anyway.